internal/pkg/cli/init.go

```diff
@@ -28,11 +26,14 @@ const defaultEnvironmentName = "test"
 
 // InitOpts holds the fields to bootstrap a new application.
 type InitOpts struct {
-	ShouldDeploy bool // true means we should create a test environment and deploy the application in it. Defaults to false.
+	// Flags unique to "init" that's not provided by other sub-commands.
+	ShouldDeploy          bool // true means we should create a test environment and deploy the application in it. Defaults to false.
+	promptForShouldDeploy bool // true means that the user set the ShouldDeploy flag explicitly.
 
 	// Sub-commands to execute.
 	initProject actionCommand
 	initApp     actionCommand
+	initEnv     actionCommand
```

Comment:
Not related to your PR, but do you think we should consider renaming `actionCommand` to `resourceCreator` or something else more indicative of the abstraction?

Reply:
The reason why I named it `actionCommand` was to contrast it compared to a `outputCommand`and stick to terminology from the [style guide](https://github.com/aws/amazon-ecs-cli-v2/blob/master/STYLE_GUIDE.md#recommended-actions). I think I see your point that it's not obvious that an `action` necessarily creates a resource.